types/service/database_service_configuration.go

```diff
@@ -226,19 +228,19 @@ type AwsRdsUsernameAndPasswordAuthConfiguration struct {
 // AwsRdsIamAuthConfiguration represents auth configuration for AWS RDS databases that use IAM authentication. You must
 // provide AWS credentials and a username. Optionally AWS CA bundle can be supplied to verify the server's certificate.
 type AwsRdsIamAuthConfiguration struct {
-	AwsCredentials
-	Username      string `json:"username"`
-	CaCertificate string `json:"ca_certificate,omitempty"`
+	AwsCredentials *common.AwsCredentials `json:"aws_credentials,omitempty"`
```

Comment:
I would prefer using just the embedded `common.AwsCredentials` here, because:

1. AWS credentials are required fields for IAM auth, so the use of pointer is not really needed here
2. the fields in `common.AwsCredentials` already prefixed with `aws_`, so we don't really need another level of nesting to distinguish the field names between `username` and `aws_profile` (vs `username` and `profile`)

Another option, if you would prefer having the nested level in JSON, could just add the `json` tag to the struct embedding:

```go
common.AwsCredentials `json:"aws_credentials"`
```

or

```go
*common.AwsCredentials `json:"aws_credentials,omitempty"`
```

The field name is optional, if it has the same name as the type.